File: src/main/scala/org/scassandra/priming/Result.scala
package org.scassandra.priming

abstract class Result(val string: String)

case object Success extends Result("success")

case object ReadTimeout extends Result("read_request_timeout")

case object Unavailable extends Result("unavailable")

case object WriteTimeout extends Result("write_request_timeout")

object Result {
  def fromString(string: String): Result = {
    string match {
      case ReadTimeout.string => ReadTimeout
      case Unavailable.string => Unavailable
      case WriteTimeout.string => WriteTimeout
      case Success.string => Success
      case _ => Success
    }
  }
}

File: java-client/src/main/java/org/scassandra/http/client/Result.java
package org.scassandra.http.client;

public enum Result {
    success,
    read_request_timeout,
    unavailable,
    write_request_timeout,
    server_error,
    protocol_error,
    bad_credentials,
    overloaded,
    is_bootstrapping,
    truncate_error,
    syntax_error,
    unauthorized,
    invalid,
    config_error,
    already_exists,
    unprepared,
    closed_connection
}
